#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_info-emacs-info-invocation.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Fixes parsing of options by the info-emacs-info script

@DPATCH@

--- octave2.1-2.9.8.orig/examples/info-emacs-info	2006-09-03 16:35:07.466100032 +0200
+++ octave2.1-2.9.8/examples/info-emacs-info	2006-09-03 16:35:23.113721232 +0200
@@ -3,12 +3,12 @@
 # Written by Kurt Hornik <Kurt.Hornik@wu-wien.ac.at> on 1996/07/01
 # Make Octave's `help -i' use Emacs info.
 # Requires a running Emacs and gnuserv.
-cmd="(Info-find-node \"$2\" \"Top\")"
-if [ $3 = "--directory" ];
+cmd="(Info-find-node \"$4\" \"Top\")"
+if [ $1 = "--directory" ];
 then
-	cmd="(add-to-list 'Info-directory-list \"$4\") $cmd"
+	cmd="(add-to-list 'Info-directory-list \"$2\") $cmd"
 	shift 2
 fi
 cmd="(require 'info) $cmd"
 [ $4 ] && cmd="$cmd (Info-index \"$4\")"
-gnudoit -q $cmd
+gnuclient -q -eval "$cmd"
